File: src/main/java/damian/tab/bankreality/database/model/Credit.java
package damian.tab.bankreality.database.model;

import lombok.*;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Getter
@Setter
public class Credit implements PrimaryKeyAccess {
    @Id
    @NotNull
    @GeneratedValue(generator = "credit_generator")
    @SequenceGenerator(name = "credit_generator", sequenceName = "credit_sequence", initialValue = 3000000, allocationSize = 10)
    private int id;

    @NotNull
    @Column
    private double loanAmount;

    @NotNull
    @Column
    private int interestRate;

    @NotNull
    @Column
    private int loanTime;

    @NotNull
    @Column
    private double EMI;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "login")
    private Client client;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "accountNumber")
    private Account account;

    @Override
    public int receivePrimaryKey() {
        return id;
    }

    @Override
    public void setPrimaryKey(int primaryKey) {
        id = primaryKey;
    }
}

File: src/main/resources/frontend/angular-app/src/app/credit/model/credit.ts
import { Client } from '../../shared/class/client/client';
import { Account } from '../../account/model/account';
export class Credit implements DbEntity<Credit>{

  id: number;
  loanAmount: number;
  interestRate: number;
  loanTime: number;
  emi: number;
  client: Client;
  account: Account;
  clientId: number;
  accountId: number;


  buildNewObject(): Credit {
    return new Credit();
  }

  setId(num: number) {
    this.id = num;
  }

}
